src/background/background.js

```diff
@@ -66,11 +49,32 @@ async function load() {
 
     await addMenuItemProcessInbox();
     await addMenuItemProcessFolder();
+};
 
-    // Add a listener for the onNewMailReceived events.
-    // On each new message decide what to do
-    // Messages are already filtered by junk classification and message filters
-    await messenger.messages.onNewMailReceived.addListener(onNewMailReceived);
-}
+// listener creation has to be on top level to work with manifest v3 
+// https://developer.chrome.com/docs/extensions/develop/migrate/to-service-workers#register-listeners
+
+// Add a listener for the onNewMailReceived events.
+// On each new message decide what to do
+// Messages are already filtered by junk classification and message filters
+messenger.messages.onNewMailReceived.addListener(onNewMailReceived);
+
+messenger.menus.onClicked.addListener(async (info, tab) => {
+    if (info.menuItemId == "catchallbirdMenuItemProcessFolder") {
```

Comment:
please make this and the other id a constant instead of two strings.  
This avoid bugs related to typos or inconsistent changes.

Reply:
Done